opm/autodiff/VFPProdProperties.hpp

```diff
@@ -0,0 +1,164 @@
+     * the above parameters from the values in the input table.
+     */
+    ADB bhp(const std::vector<int>& table_id,
+            const Wells& wells,
+            const ADB& qs,
+            const ADB& thp,
+            const ADB& alq) const;
+
+    /**
+     * Linear interpolation of bhp as a function of the input parameters given as ADBs
+     * Each entry corresponds typically to one well.
+     * @param table_id Table number to use. A negative entry (e.g., -1)
+     *                 will indicate that no table is used, and the corresponding
+     *                 BHP will be calculated as a constant -1e100.
+     * @param aqua Water phase
+     * @param liquid Oil phase
+     * @param vapour Gas phase
+     * @param thp Tubing head pressure
+     * @param alq Artificial lift or other parameter
+     *
+     * @return The bottom hole pressure, interpolated/extrapolated linearly using
+     * the above parameters from the values in the input table, for each entry in the
+     * input ADB objects.
+     */
+    ADB bhp(const std::vector<int>& table_id,
+            const ADB& aqua,
+            const ADB& liquid,
+            const ADB& vapour,
+            const ADB& thp,
+            const ADB& alq) const;
+
+    /**
+     * Linear interpolation of bhp as a function of the input parameters
+     * @param table_id Table number to use
+     * @param aqua Water phase
+     * @param liquid Oil phase
+     * @param vapour Gas phase
+     * @param thp Tubing head pressure
+     * @param alq Artificial lift or other parameter
+     *
+     * @return The bottom hole pressure, interpolated/extrapolated linearly using
+     * the above parameters from the values in the input table.
+     */
+    double bhp(int table_id,
+            const double& aqua,
+            const double& liquid,
+            const double& vapour,
+            const double& thp,
+            const double& alq) const;
+
+    /**
+     * Linear interpolation of thp as a function of the input parameters
+     * @param table_id Table number to use
+     * @param aqua Water phase
+     * @param liquid Oil phase
+     * @param vapour Gas phase
+     * @param bhp Bottom hole pressure
+     * @param alq Artificial lift or other parameter
+     *
+     * @return The tubing hole pressure, interpolated/extrapolated linearly using
+     * the above parameters from the values in the input table.
+     */
+    double thp(int table_id,
+            const double& aqua,
+            const double& liquid,
+            const double& vapour,
+            const double& bhp,
+            const double& alq) const;
+
+    /**
+     * Returns the table associated with the ID, or throws an exception if
+     * the table does not exist
+     */
+    const VFPProdTable* getTable(const int table_id) const;
+
+    /**
+     * Returns true if no vfp tables are in the current map
+     */
+    inline const bool empty() const {
```

Comment:
It has no effect to add const to a return-by-value, and therefore generates a warning. The inline should also go, it is unnecessary for methods defined inside a class (only really required for non-template functions defined in headers).